### ISSUE TITLE
App crash on notification click: `java.lang.IllegalStateException: FragmentManager has been destroyed`

### ISSUE THREAD

**Issue by tzerber** [nextcloud/android#11476]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

I'm on phone, so i really hope this copied properly. Got notification that Nextcloud found new foler with images called Picures,  clicked on the menu and got this. Never seen such issue before, and i'm using Nextcloud for quite a while. Will provide additional info if needed ;)

### Cause of error
```java
Exception in thread "main" java.lang.IllegalStateException: FragmentManager has been destroyed
    at androidx.fragment.app.FragmentManager.enqueueAction(FragmentManager.java:1644)
    at androidx.fragment.app.BackStackRecord.commitInternal(BackStackRecord.java:341)
    at androidx.fragment.app.BackStackRecord.commit(BackStackRecord.java:306)
    at androidx.fragment.app.DialogFragment.show(DialogFragment.java:279)
    at com.owncloud.android.ui.activity.SyncedFoldersActivity.onSyncFolderSettingsClick(SyncedFoldersActivity.kt:589)
    at com.owncloud.android.ui.activity.SyncedFoldersActivity$load$1$1.invokeSuspend(SyncedFoldersActivity.kt:331)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
    at android.os.Handler.handleCallback(Handler.java:938)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:210)
    at android.os.Looper.loop(Looper.java:299)
    at android.app.ActivityThread.main(ActivityThread.java:8306)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:576)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1073)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30240190`
* Build flavor: `gplay`

### Device information
* Brand: `POCO`
* Device: `camellian`
* Model: `M2103K19PG`
* Id: `SP1A.210812.016`
* Product: `camellian_p_eea`

### Firmware
* SDK: `31`
* Release: `12`
* Incremental: `V13.0.7.0.SKSEUXM`

### Expected behaviour

It should open the sync folder settings, but got this.

### Actual behaviour

App crashed and asked if i want to report the issue

### Android version

12 SP1a

### Device brand and model

Poco M3 pro 5g

### Stock or custom OS?

Stock

### Nextcloud android app version

3.24.1

### Nextcloud server version

docker 25.0.4

### Using a reverse proxy?

Yes

**Comment by abhinavs1920** [comment]
I want to work on this issue

**Comment by joshtrichards** [comment]
Fixed in #13075
